torch_geometric/utils/spmm.py

```diff
@@ -31,7 +34,10 @@ def spmm(src: Adj, other: Tensor, reduce: str = "sum") -> Tensor:
 
     :rtype: :class:`Tensor`
     """
-    assert reduce in ['sum', 'add', 'mean', 'min', 'max']
+    reduce = 'sum' if reduce == 'add' else reduce
+
+    if reduce not in ['sum', 'mean', 'min', 'max']:
+        raise ValueError(f"`reduce` argument '{reduce}' not supported")
 
     if isinstance(src, SparseTensor):
         return torch_sparse.matmul(src, other, reduce)
```

Comment:
Hi @JakubPietrakIntel  @rusty1s , May I know the logic here? If `src` is `SparseTensor` which is the default data type, it will never go into the spmm optimized impl in PT.

Reply:
Yes, I changed this back for now since `torch.sparse.mm` is missing CUDA support. I think we can either patch this in `SparseTensor` or patch this here such that we only call `torch.sparse.mm` in case PyTorch >= 2.0 and CPU. Otherwise, I think this solution is fine, as we support `torch.sparse.Tensor` now anyway (and are in the process of removing `torch-sparse` dependency).

Reply:
Understand. So for now, we need add a patch to call `torch.sparse.mm` when PT >= 2.0 and CPU. Do you know the time of adding CUDA support?

Reply:
I don‘t have any insights into that. Let me add the optimized routine in a separate PR.

Reply:
Fixed this in https://github.com/pyg-team/pytorch_geometric/pull/6759

Reply:
Thanks. The fix LGTM.